go/tools/builders/nogo_validation.go

```diff
@@ -0,0 +1,27 @@
+package main
+
+import (
+	"errors"
+	"os"
+)
+
+func nogoValidation(args []string) error {
+	validationOutput := args[0]
+	logFile := args[1]
+	// Always create the output file, but fail if the log file is non-empty, to
+	// avoid a "action failed to create outputs" error.
+	if err := os.WriteFile(validationOutput, []byte{}, 0644); err != nil {
```

Comment:
nit: we can use `os.Create` for short.

Optionally, we can write logContent to `validationOutput` to preserve the log